src/itemstackmetadata.cpp

```diff
@@ -58,15 +59,28 @@ bool ItemStackMetadata::setString(const std::string &name, const std::string &va
 	return result;
 }
 
-void ItemStackMetadata::serialize(std::ostream &os) const
+void ItemStackMetadata::serialize(std::ostream &os, bool disk) const
 {
 	std::ostringstream os2;
 	os2 << DESERIALIZE_START;
+	std::string unsent_fields;
 	for (const auto &stringvar : m_stringvars) {
+		if (! disk
+				&& stringvar.first != TOOLCAP_KEY
+				&& stringvar.first != "description"
+				&& stringvar.first != "color"
+				&& stringvar.first != "short_description"
+				&& stringvar.first != "palette_index") {
+			unsent_fields += stringvar.first + "=" + stringvar.second;
+			continue;
+		}
 		if (!stringvar.first.empty() || !stringvar.second.empty())
 			os2 << stringvar.first << DESERIALIZE_KV_DELIM
 				<< stringvar.second << DESERIALIZE_PAIR_DELIM;
 	}
+	if (! unsent_fields.empty())
+		os2 << "hash" << DESERIALIZE_KV_DELIM
```

Comment:
Special field must be documented because it would conflict with fully sent meta fields (unknown purpose of `hash`).

Suggestion: rename & document as `_hash` or similar

Reply:
How would this conflict when only tool_capabilities, description, color, short_description and palette index are sent?

Reply:
What if a mod uses `hash` in its metadata? If a server prevents map saving (edited source), there would be no way to tell these two apart.
It's about the concept. I don't expect any issues with that, but it would be cleaner to use a special reserved name for it.